src/components/EventDetail.tsx

```diff
@@ -2,36 +2,28 @@
 
 import cx from 'classnames';
 import Link from 'next/link';
-import { useRouter } from 'next/navigation';
 
-import { useAuth } from '../../../context/AuthContext';
-import { HeartButton } from '../../../components/HeartButton';
-import { DeleteIcon } from '../../../components/icons/DeleteIcon';
-import { mockEvent } from './mock';
-import { Event } from '../../../types/Event';
-import { deleteEvent } from '../../../api/events';
-import { Routes } from '../../../utils/constants';
-import { formatDate, formatTime } from '../../../utils/dateUtils';
+import { useAuth } from '../context/AuthContext';
+import { HeartButton } from './HeartButton';
+import { Event } from '../types/Event';
+import { Routes, levelLabels, sportsLabels } from '../utils/constants';
 
 type EventDetailProps = {
   event: Event;
   className?: string;
+  formattedDateTime: {
+    formattedDate: string;
+    formattedTime: string;
+  };
 };
 
 export const EventDetail: React.FC<EventDetailProps> = ({
-  event: { id, sport, location, price, date, level },
+  event: { id, sport, location, price, level, ownerId },
   className,
+  formattedDateTime: { formattedDate, formattedTime },
 }) => {
   const isUserLoggedIn = useAuth().isUserLoggedIn;
-  const router = useRouter();
-
-  const formattedDate = formatDate(date);
-  const formattedTime = formatTime(date);
-
-  const handleDelete = async () => {
-    await deleteEvent(id);
-    router.push(Routes.DASHBOARD);
-  };
+  const { id: userId } = useAuth().user;
```

Comment:
Isn't this the only thing that prevents the whole detail from being a server component? Could we move it around? Let's keep that as nice to have

Reply:
hmm...not sure where to move it 🤔 🤔

Reply:
higher or lower in the component tree

Reply:
Done